### PR TITLE
Add Site URL to third-party cookie error message

### PR DESCRIPTION
**Pull request by MatthewDorner**
#### Summary
Add Site URL to third-party cookie error message.

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-plugin-jira/issues/476

### CODE REVIEW

File: server/user_cloud.go [hunk 1]
```diff
@@ -81,11 +81,14 @@ func httpACUserInteractive(jci *jiraCloudInstance, w http.ResponseWriter, r *htt
...
 	}
+
 	mattermostUserId := r.Header.Get("Mattermost-User-ID")
 	if mattermostUserId == "" {
+		siteURL := jci.Plugin.GetSiteURL()
 		return http.StatusUnauthorized, errors.New(
 			`Mattermost failed to recognize your user account. ` +
-				`Please make sure third-party cookies are not disabled in your browser settings.`)
+				`Please make sure third-party cookies are not disabled in your browser settings. ` +
+				`Make sure you are signed into Mattermost on ` + siteURL + `.`)
```
Comment (levb) [inline server/user_cloud.go]: 👍

**Comment by mickmister** [comment]
I'd like to point out that this will work correctly on our community setup, because `community.mattermost.com` is the URL that Jira will try to connect to, no matter what. So we always hit a node that has that `SiteURL`.

**Comment by DHaussermann** [comment]
@MatthewDorner or @mickmister I'm not hitting the new (or old) message about 3rd party cookies. Do I need to set something up where a Mattermost server can be hit from multiple URLs? 

I tried testing it this way:
- Install Jira instance
- Attempt to connect to jira
- Copy URL from the new tab that opens with all of the parameters
- Paste the URL into a new session

I see no change in behavior. Jira Server shows `Failed to connect to Jira. Not authorized`   and cloud shows `invalid JWT claim sub` 
It seems this is not a valid way to test the use case we are improving?

**Comment by MatthewDorner** [comment]
I didn't look into the cause of the issue, but was told to reproduce it by going to "Site Settings" -> "Cookies and site data" -> "Block third-party cookies" in Chrome. @mickmister may be able to explain better.

**Comment by mickmister** [comment]
@DHaussermann This error string is specific to Jira Cloud. Jira Server may have the same issue so I think we should include this improvement there too.

I'm not sure what is causing the JWT error. I think I've encountered the error before but I don't remember repro steps. I suggest following the steps @MatthewDorner provided to repro the intended error.

**Comment by hanzei** [comment]
@aaronrothschild Heads up that I'm going to merge this PR despite your review missing given that it's a small change.

**Comment by hanzei** [comment]
Thanks for the quick fix @MatthewDorner :+1:

**Pull request merged**